JavaScript/task1/task1.js

```diff
@@ -0,0 +1,331 @@
+/*a*/
+
+/*Создать свою реализацию функции map для массивов*/
+
+Array.prototype.map = function(projectionFunction) {
+    const arr = [];
+    for (let i =0; i < this.length; i++){
+        arr.push(projectionFunction(this[i], i, this));
+    }
+    return arr;
+    };
+
+console.log([1,2,3].map(function(x) { return x + 1; })) === '[2,3,4]';
+
+
+/*б*/
+
+/*Переделайте массив так, чтобы объекты были следующего вида { id: …, title: … }.
+Использовать функцию map */
+
+var newReleases = [{
+    'id': 70111470,
+    'title': 'Die Hard',
+    'boxart': 'http://cdn-0.nflximg.com/images/2891/DieHard.jpg',
+    'uri': 'http://api.netflix.com/catalog/titles/movies/70111470',
+    'rating': [4.0],
+    'bookmark': []
+    }, {
+    'id': 654356453,
+    'title': 'Bad Boys',
+    'boxart': 'http://cdn-0.nflximg.com/images/2891/BadBoys.jpg',
+    'uri': 'http://api.netflix.com/catalog/titles/movies/70111470',
+    'rating': [5.0],
+    'bookmark': [{ id: 432534, time: 65876586 }]
+    }, {
+    'id': 65432445,
+    'title': 'The Chamber',
+    'boxart': 'http://cdn-0.nflximg.com/images/2891/TheChamber.jpg',
+    'uri': 'http://api.netflix.com/catalog/titles/movies/70111470',
+    'rating': [4.0],
+    'bookmark': []
+    }, {
+    'id': 675465,
+    'title': 'Fracture',
+    'boxart': 'http://cdn-0.nflximg.com/images/2891/Fracture.jpg',
+    'uri': 'http://api.netflix.com/catalog/titles/movies/70111470',
+    
+    'rating': [5.0],
+    'bookmark': [{ id: 432534, time: 65876586 }]
+    }];
+
+
+function transformArray(arr){
+    return arr.map(function({id,title}){
+        return {id, title}
+    });       
+}
+
+console.log(transformArray(newReleases));
+
+
+/*в*/
+
+/* Создать свою реализацию функции filter для массивов*/
+
+Array.prototype.filter = function(predicateFunction) {
+    let filteredArray = [];
+
+    for (let i = 0; i < this.length; i++){
+        if (predicateFunction(this[i], i, this)){
+            filteredArray.push(this[i]);
+        }      
+    }
+    return filteredArray;
+}
+
+console.log([1,2,3].filter(function(x) { return x > 2})) === "[3]";
+
+
+
+/*г*/
+/* Выведите массив ids для видео у которых рейтинг 5.0. В качестве входных данных
+используйте newReleases из предыдущих заданий. */
+
+// [675465, …]
+
+function ratingFilter(arr){
+    let filteredArray = [];
+
+    arr
+    .filter(function(obj){
+        if (obj['rating'][0] == 5.0){
+        filteredArray.push(obj['id']);
```

Comment:
.filter returns filtered array, you use it like it just iterates through array values